internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,293 @@
+	// validate the RegistryPod struct and ensure required fields are set
+	err := rp.validate()
+	if err != nil {
+		return nil, "", fmt.Errorf("error in validating RegistryPod struct: %v", err)
+	}
+
+	// create the registry pod on the cluster
+	pod, err := rp.createPodOnCluster(ctx)
+	if err != nil {
+		podLogs, logErr := rp.FetchPodLogs(ctx, pod)
+		if logErr != nil {
+			return nil, podLogs, fmt.Errorf("error creating pod: %v: and fetching logs: %v", err, logErr)
+		}
+		return nil, podLogs, fmt.Errorf("error creating pod: %v", err)
+	}
+
+	// verify that pod is successfully created
+	err = rp.validatePodOnCluster(ctx, pod)
+	if err != nil {
+		podLogs, logErr := rp.FetchPodLogs(ctx, pod)
+		if logErr != nil {
+			return nil, podLogs, fmt.Errorf("error verifying pod creation: %v: and fetching logs: %v", err, logErr)
+		}
+		return nil, podLogs, fmt.Errorf("error verifying pod creation: %v", err)
+	}
+
+	return pod, "", nil
+}
+
+// createPodOnCluster returns a pod built from an index image or returns if an existing pod with the same name is found
+func (rp *RegistryPod) createPodOnCluster(ctx context.Context) (*corev1.Pod, error) {
+	// call podForBundleRegistry() to make the pod definition
+	pod, err := rp.podForBundleRegistry()
+	if err != nil {
+		return nil, fmt.Errorf("error in building registry pod: %v", err)
+	}
+
+	// Check if Pod already exists
+	tmpPod, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Create pod in kubernetes cluster using the above pod definition and clientset
+			pod, err = rp.Kubeclient.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+			if err != nil {
+				return nil, fmt.Errorf("error creating registry pod: %v", err)
+			}
+		} else {
+			return nil, fmt.Errorf("error getting existing registry pod: %v", err)
+		}
+	} else {
+		return tmpPod, nil
+	}
+
+	return pod, nil
+}
+
+// validatePodOnCluster polls and verifies that the pod status is running
+func (rp *RegistryPod) validatePodOnCluster(ctx context.Context, pod *corev1.Pod) error {
+	// Upon creation of new pod, poll and verify that pod status is running
+	podCheck := wait.ConditionFunc(func() (done bool, err error) {
+		p, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting pod %s: %w", pod.Name, err)
+		}
+		return p.Status.Phase == corev1.PodRunning, nil
+	})
+
+	// poll every 200 ms until podCheck is true or context is done
+	err := wait.PollImmediateUntil(time.Duration(200*time.Millisecond), podCheck, ctx.Done())
+	if err != nil {
+		return fmt.Errorf("error waiting for registry pod %s to run: %v", pod.Name, err)
+	}
+
+	return err
+}
+
+// validate will ensure that RegistryPod required fields are set
+// and throws error if not set
+func (rp *RegistryPod) validate() error {
+	if len(rp.BundleImage) == 0 {
```

Comment:
```suggestion
	if len(strings.TrimSpace(rp.BundleImage)) < 1 {
```

Reply:
IMO, no need to do `strings.TrimSpace` I think. Keeping it as original.

Reply:
If we do no use the strings.TrimSpace it means that it will allow pass if the value be == "      " (empty string). Should it be a valid value and used to create the Pod?